programs/fileio.c

```diff
@@ -62,6 +57,20 @@
 #  include <lz4.h>
 #endif
 
+// Multi-OS support for determining if terminal is tty
```

Comment:
The code itself is fine.
Just, it competes with an existing and included code, which is here : 
https://github.com/facebook/zstd/blob/dev/programs/platform.h#L97

To be fair, I'm surprised the compiler doesn't complain about duplicate macro symbols (?).

Anyway, if your version is better, even in a limited way,
I believe a better outcome would be to improve the implementation in `platform.h`, so that the improvement can be shared with any other module using `IS_CONSOLE()` (such as `zstdcli.c`).